### PR TITLE
Feat: Support for loadBalancerSourceRanges on the envoy service

### PR DESCRIPTION
**Pull request by jaynis**
This PR adds support for `loadBalancerSourceRanges` on the envoy service. `loadBalancerSourceRanges` are used to specify firewall rules on the underlying load balancer of the kubernetes platform provider, if the provider supports this.

### CODE REVIEW

**Comment by jaynis** [comment]
Failing tests seem to be related to #2269. Any idea how to overcome this?

**Comment by arkodg** [comment]
@jaynis this API may make sense since the knob is specific to LB
to immediately unblock yourself, suggest using the new `patch` field
```
apiVersion: gateway.envoyproxy.io/v1alpha1
kind: EnvoyProxy
metadata:
  name: custom-proxy-config
  namespace: envoy-gateway-system
spec:
  logging:
    level:
      default: warn
  provider:
    kubernetes:
      envoyService:
        patch:
          value:
            spec:
              loadBalancerSourceRanges: 1.0.0.0/8
    type: Kubernetes
```

File: api/v1alpha1/validation/envoyproxy_validate.go [hunk 1]
```diff
@@ -115,6 +115,12 @@ func validateService(spec *egv1a1.EnvoyProxySpec) []error {
 				errs = append(errs, fmt.Errorf("allocateLoadBalancerNodePorts can only be set for %v type", egv1a1.ServiceTypeLoadBalancer))
 			}
 		}
+		if serviceType, serviceLoadBalancerSourceRanges :=
```
Comment (arkodg) [inline api/v1alpha1/validation/envoyproxy_validate.go:118]: can we add a IP Address check in here

**Comment by arkodg** [comment]
hey @jaynis still working on this one ?

**Comment by jaynis** [comment]
Sorry totally forgot about this one. What kind of IP address check do you have in mind there?

**Comment by arkodg** [comment]
> Sorry totally forgot about this one. What kind of IP address check do you have in mind there?

something like https://github.com/envoyproxy/gateway/blob/d383ba54020a1bd38e2787eff230b055713f35e3/api/v1alpha1/validation/envoyproxy_validate.go#L123

**Comment by jaynis** [comment]
I have added a CEL validation rule to check whether all `loadBalancerSourceRanges` have a valid CIDR notation. But now I am hitting some kind of schema validation rule cost budget. Limiting the maximum number of items, as suggested in the error message, does not help either.

Any suggestions how to proceed here @arkodg? Should I simply remove the CEL validation again?

**Comment by arkodg** [comment]
> schema validation rule cost budget

yah should be fine for now, you've already added logic inside the validate method

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zirain — APPROVED** [review]

**Pull request merged**